packages/waku/src/router/create-pages.ts

```diff
@@ -248,6 +248,52 @@ type ComponentList = {
 
 type ComponentEntry = FunctionComponent<any> | ComponentList;
 
+const getRoutePriority = (
+  // @TODO: maybe a better type for this?
+  pathConfig: {
+    path: PathSpec;
+    type: 'route' | 'api';
+  },
+) => {
+  const path = pathConfig.path;
+  const pathLength = path.length;
+  const hasWildcard = path.at(-1)?.type === 'wildcard';
+
+  // Calculate specificity score:
+  // - Higher path length = more specific (higher priority)
+  // - No wildcard = more specific than wildcard at the end
+  let specificityScore = pathLength * 1000; // Base score from path length
+
+  if (!hasWildcard) {
+    // No wildcard - most specific
+    specificityScore += 100;
+  } else {
+    // Wildcard at the end - less specific
+    specificityScore += 0;
+  }
```

Comment:
I don't prefer those magic numbers. You should be able to directly implement the logic in `routePriorityComparator`.